### PR TITLE
Comparison table update

### PR DESCRIPTION
**Pull request by ipdemes**
separated ufuncs from other operations

### CODE REVIEW

File: docs/cunumeric/source/comparison/_comparison_generator.py [hunk 1]
```diff
@@ -1,44 +1,62 @@
...
+
+blocklist = [
+    "test",
+    "add_docstring",
+    "abs",
+    "add_newdoc",
+    "add_newdoc_ufunc",
+    "alen",
+    "alltrue",
+    "bitwise_not",
```
Comment (magnatelee) [inline docs/cunumeric/source/comparison/_comparison_generator.py:13]: why is this excluded? we did implement `bitwise_not`.

File: docs/cunumeric/source/comparison/_comparison_generator.py [hunk 2]
```diff
@@ -1,44 +1,62 @@
...
+    "compare_chararrays",
+    "cumproduct",
+    "fastCopyAndTranspose",
+    "get_array_wrap",
+    "iterable",
+    "max",
```
Comment (magnatelee) [inline docs/cunumeric/source/comparison/_comparison_generator.py:19]: the same goes here. I guess I simply don't understand what this block list is doing.

File: docs/cunumeric/source/comparison/_comparison_generator.py [hunk 3]
```diff
@@ -1,44 +1,62 @@
...
+            and not isinstance(getattr(obj, n), type)  # not class
+            and n[0].islower()  # starts with lower char
+            and not n.startswith("__")  # not special methods
+            and not isinstance(getattr(obj, n), numpy.ufunc)
+        )
+    except:  # noqa: E722
+        return False
+
+
+def _filter_ufunc(obj, n):
```
Comment (magnatelee) [inline docs/cunumeric/source/comparison/_comparison_generator.py:51]: I guess you could factor out the common part of this and `_filter` even further. the two are different only in the last predicate.

File: docs/cunumeric/source/comparison/_comparison_generator.py [hunk 4]
```diff
@@ -62,12 +84,22 @@ def _import(mod, klass):
...
 
 
-def _section(header, mod_ext, other_lib, klass=None, exclude_mod=None):
+def _section(
+    header, mod_ext, other_lib, klass=None, exclude_mod=None, ufuncs=False
+):
     base_mod = "numpy" + mod_ext
     other_mod = other_lib + mod_ext
 
+    if ufuncs:
```
Comment (magnatelee) [inline docs/cunumeric/source/comparison/_comparison_generator.py:93]: why do you do this? `klass` and `exclude_mod` are `None` by default, so you just need to make sure you don't pass a non `None` value from the caller.

File: docs/cunumeric/source/comparison/_comparison_generator.py [hunk 5]
```diff
@@ -62,12 +84,22 @@ def _import(mod, klass):
...
+        klass = None
+        exclude_mod = None
+
+    base_funcs = []
     base_obj, base_fmt = _import(base_mod, klass)
-    base_funcs = _get_functions(base_obj)
+    if ufuncs:
```
Comment (magnatelee) [inline docs/cunumeric/source/comparison/_comparison_generator.py:99]: I guess you could pass a getter as an argument, instead of choosing it on a boolean flag.

File: docs/cunumeric/source/comparison/_comparison_generator.py [hunk 6]
```diff
@@ -131,9 +163,7 @@ def _section(header, mod_ext, other_lib, klass=None, exclude_mod=None):
         "   Number of NumPy functions: {}".format(len(base_funcs)),
         "   Number of functions covered by "
         f"{other_lib}: {len(lg_funcs & base_funcs)}",
-        "   {} specific functions:".format(other_lib),
```
Comment (magnatelee) [inline docs/cunumeric/source/comparison/_comparison_generator.py:134]: why is this removed? is it relevant to the extension done in this pr?

File: docs/cunumeric/source/comparison/_comparison_generator.py [hunk 7]
```diff
@@ -131,9 +163,7 @@ def _section(header, mod_ext, other_lib, klass=None, exclude_mod=None):
...
     ]
-    buf += ["   - {}".format(f) for f in (lg_funcs - base_funcs)]
```
Comment (magnatelee) [inline docs/cunumeric/source/comparison/_comparison_generator.py:136]: the same goes here. if this is a separate improvement you want to make, please make a separate pr.

File: docs/cunumeric/source/comparison/_comparison_generator.py (hunk 1 above)
Comment (ipdemes) [reply docs/cunumeric/source/comparison/_comparison_generator.py:13]: because these functions are not part of numpy official API: https://numpy.org/doc/stable/search.html?q=bitwise_not

Comment (ipdemes) [reply docs/cunumeric/source/comparison/_comparison_generator.py:13]: If we don't filter them out, there will be no valid link for the numpy API for them

File: docs/cunumeric/source/comparison/_comparison_generator.py (hunk 3 above)
Comment (ipdemes) [reply docs/cunumeric/source/comparison/_comparison_generator.py:51]: Sure, I will try

File: docs/cunumeric/source/comparison/_comparison_generator.py (hunk 4 above)
Comment (ipdemes) [reply docs/cunumeric/source/comparison/_comparison_generator.py:93]: I am joining it in case if someone pass not-None value from the caller. I can remove it, if you think id doesn't make sense

File: docs/cunumeric/source/comparison/_comparison_generator.py (hunk 5 above)
Comment (ipdemes) [reply docs/cunumeric/source/comparison/_comparison_generator.py:99]: Sure

File: docs/cunumeric/source/comparison/_comparison_generator.py [hunk 8]
```diff
@@ -148,6 +178,7 @@ def generate(other_lib):
         "",
     ]
     buf += _section("Module-Level", "", other_lib)
+    buf += _section("Ufuncs", "", other_lib, None, None, True)
```
[the remark this answers is not in the document]
Comment (ipdemes) [reply docs/cunumeric/source/comparison/_comparison_generator.py:181]: Sure

File: docs/cunumeric/source/comparison/_comparison_generator.py (hunk 6 above)
Comment (ipdemes) [reply docs/cunumeric/source/comparison/_comparison_generator.py:134]: This was removed because this information is not used. Do you think I should put it back?

**Comment by ipdemes** [comment]
FYI: I had to rebase this brunch on top of 22.03 to get the latest fix to cunumeric. I will resolve merge conflicts after I address all the comments.

**Comment by magnatelee** [comment]
Make sure you don't accidentally contain changes from other PRs. (looks like you do.) I think it'd be easier for you to start from `branch-22.03` and cherry-pick the changes.

**Comment by ipdemes** [comment]
@magnatelee : are we planning to move changes from 22.03 to 22.05 ? If this is the case, this PR wouldn't have any additional commits .

**Comment by magnatelee** [comment]
> @magnatelee : are we planning to move changes from 22.03 to 22.05 ? If this is the case, this PR wouldn't have any additional commits .

you said you rebased this on branch 22.03, so I assumed that's what you wanted. but it looks like your changes rely on other changes in branch 22.05, so this should be moved to 22.05, which is fine. in no case should this PR contain changes from others.

**Comment by ipdemes** [comment]
@magnatelee  :
I have just fixed the history

**Comment by ipdemes** [comment]
@magnatelee : I think I have addressed/answered all your comments. Do you have any other ?

File: docs/cunumeric/source/versions.rst [hunk 9]
```diff
@@ -7,4 +7,5 @@ Versions
 
   22.01 <https://nv-legate.github.io/cunumeric/22.01>
   22.03 <https://nv-legate.github.io/cunumeric/22.03>
+  22.05 <https://nv-legate.github.io/cunumeric/22.05>
```
Comment (ipdemes) [inline docs/cunumeric/source/versions.rst:10]: @magnatelee : I hope you don't mind this change to come with this PR.

File: docs/cunumeric/source/comparison/_comparison_generator.py (hunk 2 above)
Comment (magnatelee) [reply docs/cunumeric/source/comparison/_comparison_generator.py:19]: again, `max` and `min` are aliases to `amax` and `amin`:
```
>>> np.max
<function amax at 0x7f6129219670>
>>> np.min
<function amin at 0x7f6129219820>
```
unless we want to deviate from NumPy's behavior, I don't see a reason to blocklist these names.

File: docs/cunumeric/source/comparison/_comparison_generator.py (hunk 4 above)
Comment (magnatelee) [reply docs/cunumeric/source/comparison/_comparison_generator.py:86]: let's not generalize things until it's really necessary. unnecessary generalization just makes the code hard to maintain for others.

File: docs/cunumeric/source/comparison/_comparison_generator.py (hunk 6 above)
Comment (magnatelee) [reply docs/cunumeric/source/comparison/_comparison_generator.py:134]: No I was merely asking the logic behind removing it. If that line doesn't serve any purpose any more, I'm fine with removing it.

File: docs/cunumeric/source/versions.rst (hunk 9 above)
Comment (magnatelee) [reply docs/cunumeric/source/versions.rst:10]: Ok.

File: docs/cunumeric/source/comparison/_comparison_generator.py (hunk 2 above)
Comment (ipdemes) [reply docs/cunumeric/source/comparison/_comparison_generator.py:19]: OK, will remove them from the blocklist
